### PR TITLE
[DR-1478] Use FileProvider for layout and not found files

### PR DESCRIPTION
**Pull request by alegarciamina**
Allow setting Layout and NotFound html files by path and resolved them using FileProvider inside DocumentMiddleware.

Layout.html and NotFound.html need to be located under wwwroot, to be resolved by FileProvider.

image

image

### CODE REVIEW

File: src/MakingSense.AspNetCore.Documentation/DocumentationOptions.cs [hunk 1]
```diff
@@ -27,11 +27,13 @@ public class DocumentationOptions
 		public IFileProvider FileProvider { get; set; }
 
 		/// <summary>
-		/// Not found handling is disabled by default, set NotFoundHtmlFile to enable it.
+		/// Not found handling is disabled by default, set NotFoundHtmlPath to enable it.
 		/// </summary>
 		public bool EnableNotFoundHandling => NotFoundHtmlFile != null && NotFoundHtmlFile.Exists;
 
-		public IFileInfo NotFoundHtmlFile { get; set; }
+		public IFileInfo NotFoundHtmlFile { get; internal set; }
```
Comment (andresmoschini) [inline src/MakingSense.AspNetCore.Documentation/DocumentationOptions.cs]: This introduces a breaking change, do you think that it is possible to mark set of `NotFoundHtmlFile` and `LayoutFile` properties obsolete keep old behavior still working?

Comment (alegarciamina) [reply src/MakingSense.AspNetCore.Documentation/DocumentationOptions.cs]: You are right, I'll take a look

File: src/MakingSense.AspNetCore.Documentation/DocumentationOptions.cs [hunk 2]
```diff
@@ -42,8 +47,13 @@ public class DocumentationOptions
 
 		public TimeSpan CacheMaxAge { get; set; }
 
+		[Obsolete("Use LayoutFilePath property")]
 		public IFileInfo LayoutFile { get; set; }
 
+		public IFileInfo DefaultLayoutFile { get; internal set; }
```
Comment (andresmoschini) [inline src/MakingSense.AspNetCore.Documentation/DocumentationOptions.cs]: Do you think that is it possible to remove these properties (`DefaultNotFoundHtmlFile` and `DefaultLayoutFile`) and convert into variables in `DocumentationMiddleware.ParseLayout()` or fields in `DocumentationMiddleware`?

Comment (alegarciamina) [reply src/MakingSense.AspNetCore.Documentation/DocumentationOptions.cs]: Yes - I think it makes sense to remove both from `DocumentationOptions`.

**Comment by andresmoschini** [comment]
👍

**Pull request merged**